agora-drep/Agora/Proxy.hs

```diff
@@ -164,133 +162,121 @@ proxyScript = plam $ \authSymbol' ctx -> P.do
-                            PTxOut addr _ _ _ <- pmatch resolved
-                            PAddress cred _ <- pmatch addr
-                            PScriptCredential scriptHash <- pmatch cred
-                            pfromData scriptHash
-                      )
-                    #$ pfilter
-                    # plam
-                      ( \input -> pmatch input $ \case
-                          PTxInInfo txOutRef' _ -> txOutRef' #== txOutRef
-                      )
-                    # scriptInputs
+            -- Spending condition 6: Transaction does not include script inputs other than own input.
+            PCons ownInput scriptInputsRest <- pmatch scriptInputs
+            PNil <- pmatch scriptInputsRest
+            let ownScriptHash = P.do
+                  PTxInInfo _ resolved <- pmatch $ pfromData ownInput
+                  PTxOut addr _ _ _ <- pmatch resolved
+                  PAddress cred _ <- pmatch addr
+                  PScriptCredential scriptHash <- pmatch cred
+                  pfromData scriptHash
 
             PScriptHash rawScriptHash <- pmatch ownScriptHash
 
-            let ownCurrencySymbol = pcon $ PCurrencySymbol rawScriptHash
-
-            let mintedGAT3 = psymbolValueOf # ownCurrencySymbol # mint
-
-            PValue mintAssetMap <- pmatch mint
-            PMap mintAssetList <- pmatch mintAssetMap
-
-            let tokenMint =
-                  pfilter
-                    # plam (\mint -> pnot #$ padaSymbol #== pfromData (pfstBuiltin # mint))
-                    # mintAssetList
+            let ownCurrencySymbol = pdata $ pcon $ PCurrencySymbol rawScriptHash
 
+            -- Spending Condition 1: Transaction burns one GAT (symbol is known from script parameter)
+            -- Spending Condition 2: Spent UTxO contains GAT
             -- Spending condition 4: Transaction does not mint or burn any tokens other than V2 and V3 GATs.
-            let singleGat3Minted =
-                  ptraceInfoIfFalse "Transaction must mint a single V3 Authority token (GAT)" $
-                    mintedGAT3 #== 1
 
-            -- Previous checks verified that GAT V2 and V3 are present in the transaction, which means that
-            -- there should be exactly two currency symbols in the value
-            let onlyGatsMinted =
-                  ptraceInfoIfFalse "Transaction must not mint or burn anything else than GAT tokens" $
-                    (plength # tokenMint) #== 2
+            -- There is a bit to unpack here:
+            -- Fact that we are burning a V2 GAT combined with Condition 6 ensures that there is a
+            -- V2 GAT in the inputs. There is only one script input so it either comes from the
+            -- currently validated input OR that there is a pubkey address that had V2 GAT.
+            -- If it is the latter that means that the whole setup is already compromised so it
+            -- does not matter what happens here anyway.
```

Comment:
If you can get a hold of a GAT you can just send it to any validator with any datum anyway so the whole DAO is busted anyway. Proposals can send GATs only to validators so if you can get one that means that there is an effect that doesn't enforce buring the GAT, but that's a bug somewhere else and nothing we can do here